packages/core/src/operator/operator-runtime/listenForChallengesCallback.ts

```diff
@@ -41,25 +43,37 @@ export async function listenForChallengesCallback(challengeNumber: bigint, chall
     try {
         const graphStatus = await getSubgraphHealthStatus();
         if (graphStatus.healthy) {
-            const { sentryWalletMap, sentryKeysMap, nodeLicenseIds, mappedPools, refereeConfig } =
-                await loadOperatorKeysFromGraph_V1(operatorState.operatorAddress, challengeNumber - 1n);
-
-            await processNewChallenge_V1(challengeNumber, challenge, nodeLicenseIds, sentryKeysMap, sentryWalletMap, mappedPools, refereeConfig);
-            // check the previous challenge, that should be closed now
-            if (challengeNumber > BigInt(1)) {
-                await processClosedChallenges_V1(challengeNumber - BigInt(1), nodeLicenseIds, sentryKeysMap, sentryWalletMap);
+            // Get Referee Config from the subgraph
+            // Load current referee config from the graph
+            const refereeVersion: number = await getCurrentRefereeVersionFromGraph();
+
+            if (refereeVersion > 6) {
+                // TODO implement new function for newer versions of the Referee
+
+            } else {
+                const { sentryWalletMap, sentryKeysMap, nodeLicenseIds, mappedPools, refereeConfig } = await loadOperatorKeysFromGraph_V1(operatorState.operatorAddress, challengeNumber - 1n);
+                await processNewChallenge_V1(challengeNumber, challenge, nodeLicenseIds, sentryKeysMap, sentryWalletMap, mappedPools, refereeConfig);
```

Comment:
This seems unnecessary, we are getting the Referee version from the Refereeconfig in `loadOperatorKeysFromGraph_V1`, we should not have to load that twice do we ?

Reply:
Per our conversation, these changes have been reverted and moved to an external function!